Liuyang apollo 1.5

msbeta opened the pull request:
add opendrive adapter for hdmap;
add third party-proj.4
add third party-tinyxml

Commit:
add hdmap feature

Commit:
add hdmap feature

Commit:
add hdmap feature

Commit:
add hdmap feature

Commit:
add hdmap feature

Commit:
add hdmap feature

Commit:
add hdmap feature

Commit:
add hdmap feature

jinghaomiao committed:
modified prediction conf in pb

lianglia-apollo committed:
added st_boundary_mapper.h/cc

lianglia-apollo committed:
added st_boundary_mapper.h/cc; warning appear at build because hdmap hasn't support for LaneInfo.is_on_lane function.

Commit:
merge conflict

Commit:
add opendrive adapter

Commit:
add opendrive adapter

Commit:
add opendrive adapter

Commit:
add opendrive adapter

Commit:
add opendrive adapter

Commit:
add opendrive adapter

Commit:
fix conflict

xiaoxq committed:
Revert "Migrate LogicPoint and LogicInfo to proto."

This reverts commit 8107fcadd2e99e5f56c97972b945f18663b7e48b.

xiaoxq committed:
Revert "Migrate common::math::Vec2d to common::Vec2D proto."

This reverts commit f4b3645ed7524767660d263f0fb0a40bc56c7fa1.

xiaoxq committed:
Revert "Add apollo::common::Vec2D proto and utils."

This reverts commit 527107f6028474931ed63001649b8bfdf8972a62.

xiaoxq committed:
Use common::math::Vec2d instead of common::Vec2D in st_graph.

lianglia-apollo committed:
added speed_optimizer

lianglia-apollo committed:
added speed_optimizer

lianglia-apollo committed:
added speed_optimizer

lianglia-apollo committed:
added speed_optimizer

lianglia-apollo committed:
added speed_optimizer

lianglia-apollo committed:
added speed_optimizer

lianglia-apollo committed:
added speed_optimizer

jinghaomiao committed:
Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

jinghaomiao committed:
Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

jinghaomiao committed:
Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

jinghaomiao committed:
Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

kechxu committed:
set up the interface of prediction_map

kechxu committed:
add writing back the lane sequence probability to the feature proto

lianglia-apollo committed:
added qp_spline_st_speed_config

startcode committed:
added PointCloudAdapter specialization (#122)

Previous implementation could not correctly distinguish ros message and
protobuf message

siyangy commented on third_party/tinyxml/BUILD:
```diff
@@ -0,0 +1,18 @@
+package(default_visibility = ["//visibility:public"])
```
can you let WORKSPACE to manage third party dependency like you did for proj4?

siyangy commented on modules/map/hdmap/adapter/xml_parser/status.h:
```diff
@@ -0,0 +1,94 @@
+/* Copyright 2017 The Apollo Authors. All Rights Reserved.
```
There's already a same Status class implemented in modoules/common. Please remove this and use that instead.

siyangy commented on modules/map/hdmap/adapter/xml_parser/error_codes.h:
```diff
@@ -0,0 +1,48 @@
+/* Copyright 2017 The Apollo Authors. All Rights Reserved.
```
There's a common/error_code proto. Remove this file and put all the error codes there.

yifeijiang committed:
added mapshow. (#131)

startcode committed:
removed ptree and added proto based config for dp_poly_path

startcode committed:
removed ptree and added proto based config for dp_poly_path

startcode committed:
Resolved free move predictor conflict

startcode committed:
removed ptree and added proto based config for dp_poly_path

startcode committed:
removed ptree and added proto based config for dp_poly_path

startcode committed:
Resolved free move predictor conflict

startcode committed:
Resolved free move predictor conflict

startcode committed:
removed ptree and added proto based config for dp_poly_path

startcode committed:
removed ptree and added proto based config for dp_poly_path

startcode committed:
removed ptree and added proto based config for dp_poly_path

jinghaomiao committed:
Updated evaluator and predictor factory and manager

lianglia-apollo committed:
Devel (#134)

* added speed_optimizer

* added qp_spline_st_speed_config

* added planning/common/decision

jinghaomiao committed:
Finalized prediction framework (#138)

* modified prediction conf in pb

* added speed_optimizer

* removed ptree and added proto based config for dp_poly_path

* added speed_optimizer

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* Finalized prediction framework

startcode committed:
added dp_st_speed and qp_spline_speed, ready for migrate

startcode committed:
include qp_spline_path in build

lianglia-apollo committed:
added qp_spline_st_boundary_mapper.h/cc (#139)

kechxu committed:
implement most of prediction_map.cc

lianglia-apollo committed:
add MakeSLPoint in common/util (#141)

siyangy committed:
Fix dreamview roadgraph rendering (#137)

unacao committed:
render perception obstacles in dreamview frontend (#142)

xiaoxq committed:
Fix control test filures after ADCTranjactory proto change.

kechxu committed:
add lru cache

siyangy committed:
Avoid copying data if message_num is set as 0

siyangy committed:
Rename function and adjust style

startcode committed:
remove data_center dp_poly_path optimizer (#147)

yifeijiang committed:
added map xysl tool

xiaoxq committed:
Simple fixes to satisfy lint.

lianglia-apollo committed:
added qp_spline_st_graph

jinghaomiao committed:
Updated predictor and evaluator managers (#151)

* modified prediction conf in pb

* added speed_optimizer

* removed ptree and added proto based config for dp_poly_path

* added speed_optimizer

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* Finalized prediction framework

* added dp_st_speed and qp_spline_speed, ready for migrate

* added qp_spline_st_boundary_mapper.h/cc (#139)

* implement most of prediction_map.cc

* add MakeSLPoint in common/util (#141)

* Fix dreamview roadgraph rendering (#137)

* render perception obstacles in dreamview frontend (#142)

* Fix control test filures after ADCTranjactory proto change.

* Avoid copying data if message_num is set as 0

* Rename function and adjust style

* remove data_center dp_poly_path optimizer (#147)

* add lru cache

* Simple fixes to satisfy lint.

* Updated evaluator and predictor managers

unacao committed:
animate turn signal in dreamview frontend

siyangy committed:
HDMapImpl -> HDMap

jinghaomiao committed:
Jinghao dev (#154)

* modified prediction conf in pb

* added speed_optimizer

* removed ptree and added proto based config for dp_poly_path

* added speed_optimizer

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* Finalized prediction framework

* added dp_st_speed and qp_spline_speed, ready for migrate

* added qp_spline_st_boundary_mapper.h/cc (#139)

* implement most of prediction_map.cc

* add MakeSLPoint in common/util (#141)

* Fix dreamview roadgraph rendering (#137)

* render perception obstacles in dreamview frontend (#142)

* Fix control test filures after ADCTranjactory proto change.

* Avoid copying data if message_num is set as 0

* Rename function and adjust style

* remove data_center dp_poly_path optimizer (#147)

* add lru cache

* Simple fixes to satisfy lint.

* Updated evaluator and predictor managers

* added qp_spline_st_graph

* Updated predictor manager

siyangy committed:
Fix typo

jinghaomiao committed:
Jinghao dev (#155)

* modified prediction conf in pb

* added speed_optimizer

* removed ptree and added proto based config for dp_poly_path

* added speed_optimizer

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* added speed_optimizer

* Added lane sequence and free move predictor framework (#127)

* Fixed planning styling issues, and resolved a conflict

Fixed planning styling issues

Fixed planning styling issues

Fixed planning styling issues

Fixed some styling errors

Resolved a conflict

* Added mlp evaluator in evaluator factory

* modified prediction conf in pb

* Added free move predictor framework

* removed ptree and added proto based config for dp_poly_path

* removed ptree and added proto based config for dp_poly_path

* Resolved free move predictor conflict

* Finalized prediction framework

* added dp_st_speed and qp_spline_speed, ready for migrate

* added qp_spline_st_boundary_mapper.h/cc (#139)

* implement most of prediction_map.cc

* add MakeSLPoint in common/util (#141)

* Fix dreamview roadgraph rendering (#137)

* render perception obstacles in dreamview frontend (#142)

* Fix control test filures after ADCTranjactory proto change.

* Avoid copying data if message_num is set as 0

* Rename function and adjust style

* remove data_center dp_poly_path optimizer (#147)

* add lru cache

* Simple fixes to satisfy lint.

* Updated evaluator and predictor managers

* added qp_spline_st_graph

* Updated predictor manager

* Restored parameters.yml

kechxu committed:
Implement obstacle::set_current_lane and add AngleDiff in apollo/common/math/math_utils.cc (#157)

wanglei828 committed:
Fix gnss build.

startcode committed:
moved STPoint to class implementation (#159)

* easier for coding
* less error prone for building ST graph.

Commit:
perception changes:
1. first subnode
2. refactor perception main

jmtao committed:
convert dp_st_speed optimizer, to make it compilable

Commit:
add hdmap interface

Commit:
add hdmap interface

Commit:
add hdmap interface

Commit:
add hdmap interface

Commit:
merge conflict

FangzhenLi-hust committed:
add lidar_process and hdmap interface (#162)

* add pcl_util

* add perception common lib files

* add lidar_process, and modify code style

siyangy commented:
@msbeta Can you clean up commit history a bit? It doesn't look right at the moment.

Commit:
add minbox builder

ycool committed:
add map into data center.

startcode committed:
moved  DPPolyPathOptimizer to dp_poly_path folder

lianglia-apollo committed:
uniform decision usage in planning. remove redundancy.

jinghaomiao committed:
Added comments for mlp evaluator header file

jinghaomiao committed:
Added is on lane check for obstacle

jinghaomiao committed:
Added pedestrian predictor framework

fengqikai1414 committed:
add tf broadcaster nodelet in gnss driver

yifeijiang committed:
added localization in mapshow

ycool committed:
switch planning to em_planner and fix diagnostics.

kechxu committed:
Partially implemented obstacle container in prediction module

startcode committed:
use decision type defined in decision proto for qp_spline_path

xiaoxq committed:
Fix some lint errors. (#174)

yifeijiang committed:
added lint check for dp_poly_path

lianglia-apollo committed:
added functions in qp_spline_st_boundary_mapper.h/cc

kechxu committed:
Implement insert pose in prediction.cc

kechxu committed:
refactor the framework of setting lane graph feature

siyangy committed:
Reorganize dreamview directory hierarchy

jinghaomiao committed:
Modified insert of pose as a perception obstacle (#184)

fengqikai1414 committed:
update ros release for 1.5 (#185)

jinghaomiao committed:
Removed unnecessary function (#186)

startcode committed:
added reference line smoother

kechxu committed:
implement trim() in obstacle.cc

lianglia-apollo committed:
uniform the interface of optimizers

siyangy committed:
migrate dreamview to apollo_app

lianglia-apollo committed:
added more functions in st_boundary_mapper. replace size_t with uint32_t

kechxu committed:
insert lane sequences into obstacle.cc in prediction module (#191)

kechxu committed:
implement set lane points in obstacle.cc in prediction module

jinghaomiao committed:
Merged with Apollo Master (#193)

* batch fix coding style problems (#243)

* fix spurious build success due to BUILD file syntax error

* Add cpplint rules during ./apollo.sh lint (#249)

jmtao committed:
(1)migrate object_table; (2)get rid of temp migration TODOs

lianglia-apollo committed:
added map_boundary_by_prediction_trajectory.

lianglia-apollo committed:
added timestamp in vehicle_state.

startcode committed:
fix qp_oases compile problem

startcode committed:
added reference line smoother test

varified that the smoothing algorithm works

jinghaomiao committed:
Updated map APIs

siyangy committed:
Remove redundant lru_cache

ycool committed:
add optimizer config. (#201)

kechxu committed:
implement set nearby lane features in obstacle.cc

lianglia-apollo committed:
changed vehicle_state to singleton.

xiaoxq committed:
Plannning optimizers return Status instead of ErrorCode.

jinghaomiao committed:
Added free move trajectory

lianglia-apollo committed:
removed vehicle_state_proxy from planning/proxy.

jinghaomiao committed:
Fixed some errors in free move trajectory

Commit:
Update docker file and image with Caffe support

ycool committed:
enable more optimizers in em_planner.

Capri2014 committed:
Replace qpSMOSolver with qpOASES

Capri2014 committed:
Polish and fix MPC/QP solver

startcode committed:
rename hdmap::PathPoint to hdmap::MapPathPoint

to avoid name confusion with type apollo::common::PathPoint defined in
path_point.proto

ycool committed:
integrate reference line smoother into em planner.

ycool committed:
fix routing file loading problem.

lianglia-apollo committed:
added trajectory stitcher

Commit:
adding min_box_test (#194)

Commit:
add new interface for hdmap

Commit:
add new interface for hdmap

Commit:
conflict merge

Commit:
fix merge problem
